### PR TITLE
Add config schema validation

### PR DESCRIPTION
**Pull request by tomphp**
# Motivation and Context
Add config validation so that missing config values are detected at start up - rather have the app generated mysterious errors during runtime if something is missing/wrong.

# What has changed
- Added the `schema` dependency
- Added a schema to `config.py`
- Added ad-hoc schema validation to the end of `config.py`

# Questions
1. Do we like this concept?
1. Do we like the `schema` library?
1. What's the best way to organise the code?

### CODE REVIEW

File: config.py [hunk 1]
```diff
@@ -136,3 +143,83 @@ class TestingConfig(DevelopmentConfig):
...
+
+URL_SCHEMA = Regex(r'^https?://')
+NON_EMPTY_STRING_SCHEMA = And(str, len)
+
+CONFIG_SCHEMA = Schema({
+    'DEBUG': bool,
+    'TESTING': bool,
+    Optional('DEVELOPMENT'): bool,
+    'VERSION': str,
+    'PORT': int,
```
[the remark this answers is not in the document]
Comment (tomphp) [reply config.py]: Ah, sorry! I've not actually finished trying to make it work - I just wanted to put it up for scrutiny to see if it's worth carrying on (added `WIP` tag)?

[the remark this answers is not in the document]
Comment (ajmaddaford) [reply config.py]: No worries, I was only running it to find out what the logging looked like when it fails.

[the remark this answers is not in the document]
Comment (tomphp) [reply config.py]: It's messy but it's something 😉

**Comment by ajmaddaford** [comment]
I think it's worth considering separating the config schema from the app. Heroku define it in their app.json - https://devcenter.heroku.com/articles/app-json-schema#env. This would give the advantages:
- We could have the same approach regardless of service implementation language
- It's easily readable and can include documentation for each environment variable
- It could be used to render documentation for services by reading repo content
- It could be used parsed easily (e.g. using [jq](https://stedolan.github.io/jq/)) and used by scripts

**Comment by NickGrantONS** [comment]
We are thinking about closing this PR, because it's quite old and there are unresolved comments

**Pull request closed without merging**